### ISSUE TITLE
to_pil_image assumes images to be 0-1 normalized if dtype == float

### ISSUE THREAD

**Issue by Isalia20** [pytorch/vision#8782]
### 🐛 Describe the bug

This has been bothering me for a while, so want to submit patches to fix it. If function `to_pil_image` receives float as the input it assumes that it needs to rescale it from 0-1 to 0-255, no matter the range of the actual values in the tensor. Example:
```
import torchvision.io as tvio
from torchvision.transforms.functional import to_pil_image

image = tvio.read_image("image.png") # 0-255 scale
image = image.float() # or .half, leads to same thing
to_pil_image(image).save("a.png") # image is corrupted
```

If you agree, I can work on the patch later this week

### Versions

Not really needed, latest version of torchvision==0.20.1 also has this
```[tasklist]
### Tasks
```

**Comment by NicolasHug** [comment]
Hi @Isalia20 ,  that's expected, and documented: https://pytorch.org/vision/stable/transforms.html#dtype-and-expected-value-range

Changing this assumption would be breaking a lot of existing code, and like all conventions, it's not necessarily worse or better than any other. Various other popular libraries like matplotlib make the same assumption.

What other strategy would you have in mind here?

**Comment by NicolasHug** [comment]
BTW:

> image = tvio.read_image("image.png") # 0-255 scale
> image = image.float() # or .half, leads to same thing

It is for this reason that users should be using `ToDtype(..., scale=True)`  to properly convert the dtype as well as the range.

**Comment by Isalia20** [comment]
Yeah that makes sense, now that I think of it. I was thinking of doing a simple check whether max of the tensor is > 1 and then assume that its in 0-255 range but there will be lots of edge cases. Closing this, thanks for the feedback
